qemu/tests/qemu_guest_agent.py

```diff
@@ -4168,15 +4185,52 @@ def _base_on_bg_check_resource_leak():
 
             if bg.is_alive():
                 qga_handles, qga_memory = _process_qga_resource()
-                # Generally the qga handles won't exceed 300, and the memory
-                # setting '3000' is an average value when the guest is not
-                # running redundant operations and apps.
-                if qga_handles > 300 or qga_memory > 3000:
-                    test.fail("QGA commands caused resource leak. handles are"
-                              " %s, memory is %s" % (qga_handles, qga_memory))
-                else:
-                    LOG_JOB.info("Current qga handles is %s, qga memory"
-                                 "is %s", qga_handles, qga_memory)
+                if qga_memory > qga_mem_threshold:
+                    qga_memory_previous = qga_memory
+                    qga_handles_previous = qga_handles
+                    qga_handles_latest, qga_memory_latest = _process_qga_resource()
+                    memory_high_peak = []
+                    memory_low_peak = []
+                    memory_list = []
+                    start_time = time.time()
```

Comment:
Let's have a counter, if the value does not change for a certain time, we might add it to memory_high_peak or memory_low_peak accordingly.
```
stable_counter = 0
```

Reply:
after further consideration and testing results, this condition way is the more reasonable and measurable way.